src/main/java/org/janelia/saalfeldlab/paintera/control/IdSelector.java

```diff
@@ -88,19 +186,19 @@ public void accept(final MouseEvent e)
 				viewer.displayToGlobalCoordinates(access);
 				final IntegerType<?> val = access.get();
 				final long id  = val.getIntegerLong();
-				actOn(id, selectedIds);
+				actOn(id);
 		}
 
-		protected abstract void actOn(final long id, SelectedIds selectedIds);
+		protected abstract void actOn(final long id);
 	}
 
 	private class SelectFragmentWithMaximumCount extends SelectMaximumCount
 	{
 
 		@Override
-		protected void actOn(final long id, final SelectedIds selectedIds)
+		protected void actOn(final long id)
 		{
-			if (Label.regular(id))
+			if (foregroundCheck.test(id))
```

Comment:
Does this mean we cannot (click-)select background anymore? I am ok with that even though sometimes the background can look cool :smile:

Reply:
Yes, from what I've seen, it happens too often that one accidentally selects the background when trying to click a small object (or clicking on its edge). We can make this a setting whether the background is selectable or not, but I can't think of any use-case where it would be needed

Reply:
Excluding foreground for this is fine with me.